packages/core/src/controllers/SnackController.ts

```diff
@@ -27,25 +27,34 @@ export const SnackController = {
   },
 
   showLoading(message: SnackControllerState['message']) {
-    state.message = message
-    state.variant = 'loading'
-    state.open = true
+    this._showMessage(message, 'loading')
   },
 
   showSuccess(message: SnackControllerState['message']) {
-    state.message = message
-    state.variant = 'success'
-    state.open = true
+    this._showMessage(message, 'success')
   },
 
   showError(message: unknown) {
     const errorMessage = CoreHelperUtil.parseError(message)
-    state.message = errorMessage
-    state.variant = 'error'
-    state.open = true
+    this._showMessage(errorMessage, 'error')
   },
 
   hide() {
     state.open = false
+  },
+
+  _showMessage(message: SnackControllerState['message'], variant: SnackControllerState['variant']) {
+    if (state.open) {
+      state.open = false
```

Comment:
Is this ok? wouldn't this close the current notification "instantly" ?  Should there be a delay before closing it?

Reply:
The open/close logics and animations are handled on `w3m-snackbar` component. When we set it to false here, it means it'll call the closing animation which will take `150ms` and new Snackbar will be introduced